card_records_spec.rb

```diff
@@ -0,0 +1,294 @@
+require 'rails_helper'
+
+RSpec.describe NewMexico::CardRecords, type: :service do
+  let(:file_prefix) { "file://#{Rails.root}/" }
+
+  after(:each) do
+    subject.close_browser
+  end
+
+  context "Other" do
+    describe "#goto" do
+      it "goes to certain url" do
+        url = "#{file_prefix}#{file_fixture("Stage_1.htm")}"
+
+        subject.goto(url)
+
+        expect(subject.send(:browser).url).to eq url
+      end
+    end
+
+    describe "#uri" do
+      it "returns correct uri" do
+        correct_uri = "https://wwwapps.emnrd.state.nm.us/ocd/ocdpermitting/OperatorData/PermitStatusParameters.aspx"
+
+        expect(subject.uri).to eq correct_uri
+      end
+
+      it "returns string" do
+        expect(subject.uri).to be_a String
+      end
+    end
+
+    describe "#filename" do
+      it "returns the output file name" do
+        file_url = "http://ocdimage.emnrd.state.nm.us/imaging/filestore/santafe/wf/20190716/30025053060000_07_16_2019_03_01_48.pdf"
+
+        filename =
+          subject.filename(
+            file_url: file_url,
+            document_type: "Type",
+            permit_id: "Permit_id",
+            filing_date: "2019-07-17"
+          )
+
+        expect(filename).to eq "30025053060000_07_16_2019_03_01_48_Type_Permit_id_2019-07-17.pdf"
+      end
+    end
+
+    describe "download_pdf" do
+      let(:url) { "#{Rails.root}/#{file_fixture("005842AD.pdf")}" }
+
+      it "downloads pdf file" do
+        file = subject.download_pdf(url)
+
+        expect(file.size).to_not be_nil
+        expect(file.size).to_not be_zero
+      end
+
+      context "when download was failure" do
+        it "raises error" do
+          allow(subject).to receive(:open).and_raise("Error")
+
+          expect { subject.download_pdf(url) }.to raise_error "Error"
+        end
+      end
+    end
+
+    describe "#upload_pdf" do
+      it "uploads pdf to S3" do
+        filename = "30025053060000_07_16_2019_03_01_48_Type_Permit_id_2019-07-17.pdf"
+        file = double("file", read: "str")
+
+        expect(subject.send(:client)).to receive(:put_object).at_least(:once) do |args|
+          expect(args[:key]).to end_with filename
+          expect(args[:content_type]).to eq "application/pdf"
+        end
+
+        subject.upload_pdf(file, filename)
+      end
+    end
+  end
+
+  context "Stage 1" do
```

Comment:
I did so because some methods use a specific page, right. I think that this reduced the number of lines of code by declaring a 'let' with the correct page of fixtures.